scripts/generate.js

```diff
@@ -0,0 +1,144 @@
+    .then((res) => res.data)
+    .then((result) => {
+      return Object.values(result)
+        .filter(Boolean)
+        .filter(entity => entity.bullhornEntitySchema && entity.bullhornEntitySchema.restAccess && entity.bullhornEntitySchema.restAccess.indexOf("READ") > -1)
+        .sort(sortByField("entityName"));
+    });
+};
+
+const generate = async (config = {}) => {
+  try {
+    let dir = config.directory || "./typings";
+    Logger.log(`retrieving metadata for each entity...`);
+    let models = await getEntityModels();
+    Logger.log(`parsing entity models...`);
+    const types = models.map((model) => parseModelData(model));
+
+    Logger.log(`writing templates...`);
+    let deletables = types.filter(
+      (i) => i && i.properties.find((p) => p.name === "isDeleted")
+    );
+    let tmp = CORE_DEFINITION_TEMPLATE({
+      types: types,
+      deletables: deletables,
+    });
+    if (!fs.existsSync(dir)) {
+      fs.mkdirSync(dir, { recursive: true });
+    }
+    fs.writeFile(`${dir}/index.ts`, tmp, (err) => {
+      if (err) {
+        Logger.error(err.message);
+      }
+      Logger.success(`Done!`);
+    });
+  } catch (err) {
+    Logger.error("✗ An error occurred", err);
+  }
+};
+
+const parseModelData = (model) => {
+  if (!model) {
+    Logger.error("Missing entity model");
+    return;
+  }
+
+  Logger.log(`${model.entityName}`);
+  let data = {
+    type: model.entityName,
+    properties: [],
+    dependencies: new Set(),
+    dynamic: false,
+  };
+  let hasCustomObjects = false;
+  if (model.entityName.indexOf("CustomObject") >= 0) {
+    data.dynamic = true;
+  } else {
+    for (let field of Object.values(model.fields).sort(
+      sortByField("fieldName")
+    )) {
+      if (["OneToMany", "ManyToMany"].includes(field.associationType)) {
+        data.properties.push({
+          name: field.fieldName,
+          type: `ToMany<${field.associatedEntityType}>`,
+        });
+        data.dependencies.add(field.associatedEntityType);
+      } else if (["ManyToOne", "OneToOne"].includes(field.associationType)) {
+        data.properties.push({
+          name: field.fieldName,
+          type: `${field.associatedEntityType}`,
+        });
+        data.dependencies.add(field.associatedEntityType);
+      } else if (["id"].indexOf(field.fieldName) < 0) {
+        switch (field.fieldType || "") {
+          case "Integer":
+          case "Double":
+          case "Float":
+          case "BigDecimal":
+          case "Long":
+            data.properties.push({
+              name: field.fieldName,
+              type: "number",
+            });
+            break;
+          case "Date":
+          case "Timestamp":
+            data.properties.push({
+              name: field.fieldName,
+              type: "Date",
+            });
+            break;
+          case "String":
+          case "byte[]":
+            data.properties.push({
+              name: field.fieldName,
+              type: "Strings",
```

Comment:
The only fields that require the `Strings` type of `string or string[]` are fields that start with customText. Configuring a field outside of a custom text field to be a multi-value string array in field maps would be either extremely rare, or a misconfig. For instance, the status string field could be configured to be a multi-value by a customer, but that would break any workflows that rely on status being a single value string. We can add logic that only applies the `Strings` to fields that are of the format: `customText[n]`. We don't need to apply this to `customTextBlock[n]` because those fields cannot be configured to be multi-value.